base/essentials.jl

```diff
@@ -158,6 +158,22 @@ macro boundscheck(blk)
       $(Expr(:boundscheck,:pop)))
 end
 
+"""
+    @inbounds(blk)
+
+In the example below the bound check of array A is skipped to improve performance.
+```
+function sum(A::AbstractArray)
+    r = zero(eltype(A))
+    for i = 1:length(A)
+        @inbounds r += A[i]
+    end
+    return r
+end
+```
+Warning: Using @inbounds may return incorrect results/clashes/corruption
```

Comment:
Perhaps use an [admonition](http://docs.julialang.org/en/stable/manual/documentation/#admonitions) here to make this stand out a bit more.

Also, the word `@inbounds` should be wrapped in single backticks to be displayed correctly as a code literal, like so: `` `@inbounds` ``.